mypy/config_parser.py

```diff
@@ -55,9 +55,12 @@ def parse_version(v: str | float) -> tuple[int, int]:
 def try_split(v: str | Sequence[str], split_regex: str = "[,]") -> list[str]:
     """Split and trim a str or list of str into a list of str"""
     if isinstance(v, str):
-        return [p.strip() for p in re.split(split_regex, v)]
-
-    return [p.strip() for p in v]
+        items = [p.strip() for p in re.split(split_regex, v)]
+    else:
+        items = [p.strip() for p in v]
+    if items and items[-1] == "":
+        items.pop(-1)
```

Comment:
can we only do this if it's a str? e.g. if we have a toml list and the last element is an empty str that feels pretty explicit and i would want an error